src/probability_theory/probability_space.lean

```diff
@@ -0,0 +1,777 @@
+/-
+Copyright (c) 2021. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Zinkevich (modified for mathlib by Hunter Monroe)
+ -/
+
+import measure_theory.measurable_space
+import measure_theory.measure_space
+
+/-!
+This file defines the basic concepts in probability theory.
+probability_space: a measure_space where the measure has a value of 1.
+event: a set that is measurable (defined based upon the measurable space),
+Pr[E]: the probability of an event.
+-/
+
+open measure_theory measurable_space
+open_locale big_operators classical
+
+namespace probability_theory
+
+/- An event is a measurable set in a measurable space that has a probability measure. Note:
+the definition below does not require that the measurable_space have volume one. -/
+@[derive [has_union,has_emptyc,has_compl, has_sdiff,has_union,has_inter,boolean_algebra]]
```

Comment:
You are deriving both `has_union` (twice) and `boolean_algebra`, which upplies `has_inf` and `has_sup`. I recommend that you don't ever use `\empty` or `\cup` and only use `\bot` and `\sqcup`. Having two ways to write something means many more unnecessary lemmas, and as Eric says if you have the boolean algebra lemmas then you can just use lemmas like `inf_assoc` directly instead of proving your own version.

Reply:
Agreed.